doc/scss/custom.scss

```diff
@@ -190,3 +190,65 @@ div.sk-text-image-grid-small {
 div.sk-text-image-grid-large {
   @include sk-text-image-grid(100px);
 }
+
+/* Style of the docsearch modal window */
+
+// In the following, we mainly reuse the background color and font-color from the
+// pydata-sphinx-theme. We also adjust slightly the font-size to be wider and in line
+// with the sidebar design.
+:root {
+  --docsearch-modal-width: 768px;
+  --docsearch-modal-height: 80vh;
+  --docsearch-searchbox-shadow: inset 0 0 0 2px var(--pst-color-primary);
+  --docsearch-highlight-color: var(--pst-color-primary);
+}
+
+.DocSearch-Button-Key {
+  top: 0;
+}
+
+@media screen and (max-width: 1200px) {
+  .DocSearch-Button-Placeholder,
+  .DocSearch-Button-Keys {
+    display: none;
+  }
+}
+
+.DocSearch-Modal {
+  background: var(--pst-color-background);
+  margin-top: 10vh;
+}
+
+.DocSearch-Form {
+  background: var(--pst-color-background);
+}
+
+.DocSearch-MagnifierLabel,
+.DocSearch-Reset {
+  color: var(--pst-color-primary);
+}
+
+.DocSearch-Input {
+  color: var(--pst-color-text-base);
+  font-size: var(--pst-font-size-icon);
+}
+
+.DocSearch-Hit-icon svg {
+  vertical-align: unset;
+}
+
+.DocSearch-Hit-source {
+  background: var(--pst-color-background);
+  color: var(--pst-color-primary);
+  font-size: var(--pst-sidebar-font-size);
+}
+
+.DocSearch-Hit-title {
+  font-size: var(--pst-sidebar-font-size);
+}
+
+// Remove contrast of footer since only Algolia logo is displayed
+.DocSearch-Footer {
+  background: var(--pst-color-background);
+  box-shadow: 0 0 0 0;
+}
```

Comment:
```suggestion
@media screen and (min-width: 768.1px) {
  .DocSearch-Modal {
    margin-top: 10vh;
  }
}

.DocSearch-Modal,
.DocSearch-Form,
.DocSearch-Hit-source {
  background: var(--pst-color-background);
}

.DocSearch-MagnifierLabel,
.DocSearch-Reset,
.DocSearch-Hit-Source {
  color: var(--pst-color-primary);
}

.DocSearch-Input {
  color: var(--pst-color-text-base);
  font-size: var(--pst-font-size-icon);
}

.DocSearch-Hit-icon svg {
  vertical-align: unset;
}

.DocSearch-Hit-source,
.DocSearch-Hit-title {
  font-size: var(--pst-sidebar-font-size);
}
```

This:

- Simplify a bit by merging same styling into one block. Most places we are tweaking the same color/background.
- `.DocSearch-Modal { margin-top: 10vh; }` should be applied only to large screens as algolia specifies full viewheight for the modal on smaller screens.
- Remove the `.DocSearch-Footer` styling as we now keeps the instructions and the comment above no longer holds true.